File: junha/algorithm-boj/week1/10866.py
from sys import stdin 

N = int(input())

deque = [] 
for _ in range(N):
    user_input = stdin.readline()
    user_command = user_input.split()[0]
    
    if user_command == 'push_front':
        x = int(user_input.split()[1])
        deque.insert(0, x)

    elif user_command == 'push_back':
        x = int(user_input.split()[1])
        deque.append(x)

    elif user_command == 'pop_front':
        print(-1 if len(deque) == 0 else deque.pop(0))

    elif user_command == 'pop_back':
        print(-1 if len(deque) == 0 else deque.pop())

    elif user_command == 'size':
        print(len(deque))

    elif user_command == 'empty':
        print(1 if len(deque) == 0 else 0)

    elif user_command == 'front':
        print(-1 if len(deque) == 0 else deque[0])

    elif user_command == 'back':
        print(-1 if len(deque) == 0 else deque[-1])

    else:
        raise NotImplementedError

File: minseok/10866/10866/10866.cpp
#include <iostream>
#include <string>
#include <deque>

using namespace std;

int main() {

	int n;
	int data;
	deque<int> DQ;
	string command;
	cin >> n;

	for (int i = 0; i < n; i++) {
		cin >> command;

		if (command == "push_front") {
			cin >> data;
			DQ.push_front(data);
		}

		else if (command == "push_back") {
			cin >> data;
			DQ.push_back(data);
		}

		else if (command == "pop_front") {
			if (DQ.empty()) cout << "-1" << endl;
			else {
				cout << DQ.front() << endl;
				DQ.pop_front();
			}
		}

		else if (command == "pop_back") {
			if (DQ.empty()) cout << "-1" << endl;
			else {
				cout << DQ.back() << endl;
				DQ.pop_back();
			}
		}

		else if (command == "size") {
			cout << DQ.size() << endl;
		}

		else if (command == "empty") {
			if (DQ.empty()) cout << 1;
			else cout << -1;
		}

		else if (command == "front") {
			if (DQ.empty()) cout << "-1" << endl;
			else cout << DQ.front();
		}

		else if (command == "back") {
			if (DQ.empty()) cout << "-1" << endl;
			else cout << DQ.back();
		}
	}
	return 0;
}